lib/internal/fs/sync_write_stream.js

```diff
@@ -23,8 +23,12 @@ ObjectSetPrototypeOf(SyncWriteStream.prototype, Writable.prototype);
 ObjectSetPrototypeOf(SyncWriteStream, Writable);
 
 SyncWriteStream.prototype._write = function(chunk, encoding, cb) {
-  writeSync(this.fd, chunk, 0, chunk.length);
-  cb();
+  try {
+    writeSync(this.fd, chunk, 0, chunk.length);
+    cb();
+  } catch (e) {
+    cb(e);
```

Comment:
If not callback, writeStream.onwrite will not call, writeState.writing is always true. So next write will push to writeState.buffered, that's why memory leak.

If callback with error, writeStream will be destroyed, and next write will throw error.

Reply:
The logic is slightly wrong though. If `cb()` on line 28 throws an exception, it ends up getting called again on line 30.

Reply:
You're right. Callback may has side effect, it should remove from the try/catch.